airflow/executors/kubernetes_executor_utils.py

```diff
@@ -143,6 +143,10 @@ def _run(
             self.log.debug("Event: %s had an event of type %s", task.metadata.name, event["type"])
             if event["type"] == "ERROR":
                 return self.process_error(event)
+            labels = task.metadata.labels
+            if labels.get("airflow-worker", None) != scheduler_job_id:
+                last_resource_version = task.metadata.resource_version
```

Comment:
Why do we need to update the `last_resource_version` when the pod is not managed by this executor? To be safe, I think we should directly `continue` when the label is not equals to the current scheduler job id. WDYT?

Reply:
The scheduler don't want to receives the same event again and again. To avoid this we have to update the last_resource_version. So, it will get the events from the last_resource_version onwards.